### DISCUSSION TITLE
how to set SlicGrid decimal places dynamicaly?

### DISCUSSION THREAD

**Discussion by rahooftkh** [serenity-is/Serenity#6413] [Help]
hi
 how to set SlicGrid decimal places dynamicaly ? there have any GridOptions

**Comment by edwardch** [comment]
In your *Row.cs, or your *Column.cs you can add this to your field:

`[DisplayFormat("#,##0.00")]`

For numbers you might also want to use `[AlignRight]`

You can create your own formatter too if you like. Create a TypeScript file like this:

```ts
namespace YourProject.YourModule  {
    @Serenity.Decorators.registerFormatter()
    export class PercentageFormatter implements Slick.Formatter {
        format(ctx: Slick.FormatterContext) {
            if (ctx.value) {
                let fmtString = `#,##0.${Q.zeroPad(0, this.decimalPlaces)}`;
                return Q.htmlEncode(Q.formatNumber(ctx.value * 100, fmtString)) + "%";
            }
        }

        @Serenity.Decorators.option()
        public decimalPlaces: number = 2
    }
}
```

Run your project to transform any of these templates.

Then add the following to your *Column.cs field:

```cs
[PercentageFormatter(DecimalPlaces = 4)]
public Decimal YourField { get; set; }
```

HTH
